flexeval/core/language_model/openai_api.py

```diff
@@ -108,7 +133,10 @@ def batch_complete_text(
                 **kwargs,
             ),
         )
-        return [res.choices[0].message.content for res in api_responses]
+        completions = [res.choices[0].message.content for res in api_responses]
+        if all(completion == "" for completion in completions):
+            logger.warning("All generated texts are empty string. Something may go wrong.")
+        return completions
```

Comment:
I think this could be moved inside `_retry_on_error`.
Just one failed response is worth raising a warning!

Reply:
Ah, I overlooked it.
LGTM 👍